shared/src/ledger/native_vp/ethereum_bridge/bridge_pool_vp.rs

```diff
@@ -118,43 +149,149 @@ where
 
     /// Check that the correct amount of Nam was sent
     /// from the correct account into escrow
+    #[inline]
     fn check_nam_escrowed(&self, delta: EscrowDelta) -> Result<bool, Error> {
+        self.check_nam_escrowed_balance(delta)
+            .map(|balance| balance.is_some())
+    }
+
+    /// Check that the correct amount of Nam was sent
+    /// from the correct account into escrow, and return
+    /// the updated escrow balance.
+    fn check_nam_escrowed_balance(
+        &self,
+        delta: EscrowDelta,
+    ) -> Result<Option<AmountDelta>, Error> {
         let EscrowDelta {
             payer_account,
             escrow_account,
             expected_debit,
             expected_credit,
         } = delta;
-        let debited = self.account_balance_delta(payer_account);
-        let credited = self.account_balance_delta(escrow_account);
+        let debit = self.account_balance_delta(payer_account);
```

Comment:
Take it or leave it, but the golfer in me thinks you should return `(Amount, SignedAmount)` to simplify the pattern matches below and only construct the `AmountDelta` from the two as needed.